airflow/providers/google/cloud/operators/compute.py

```diff
@@ -68,6 +75,468 @@ def execute(self, context: 'Context'):
+            raise AirflowException(
+                f"The body '{self.body}' should contain at least machine type for the new operator "
+                f"in the 'machine_type' field. Check (google.cloud.compute_v1.types.Instance) "
+                f"for more details about body fields description."
+            )
+        if 'disks' not in self.body:
+            raise AirflowException(
+                f"The body '{self.body}' should contain at least disks for the new operator "
+                f"in the 'disks' field. Check (google.cloud.compute_v1.types.Instance) "
+                f"for more details about body fields description."
+            )
+        if 'network_interfaces' not in self.body:
+            raise AirflowException(
+                f"The body '{self.body}' should contain at least network interfaces for the new operator "
+                f"in the 'network_interfaces' field. Check (google.cloud.compute_v1.types.Instance) "
+                f"for more details about body fields description. "
+            )
+
+    def _validate_all_body_fields(self) -> None:
+        if self._field_validator:
+            self._field_validator.validate(self.body)
+
+    def _validate_inputs(self) -> None:
+        if self.project_id == '':
+            raise AirflowException("The required parameter 'project_id' is missing")
+        if not self.zone:
+            raise AirflowException("The required parameter 'zone' is missing")
+        if not self.resource_id and "name" not in self.body:
+            raise AirflowException(
+                "The required parameters 'resource_id' and  body['name'] are missing. "
+                "Please, provide at least one of them."
+            )
+
+    def execute(self, context: 'Context') -> dict:
+        hook = ComputeEngineHook(
+            gcp_conn_id=self.gcp_conn_id,
+            api_version=self.api_version,
+            impersonation_chain=self.impersonation_chain,
+        )
+        self._validate_all_body_fields()
+        self.check_body_fields()
+        try:
+            # Idempotence check (sort of) - we want to check if the new Instance
+            # is already created and if is, then we assume it was created previously - we do
+            # not check if content of the Instance is as expected.
+            # We assume success if the Instance is simply present.
+            existing_instance = hook.get_instance(
+                resource_id=self.resource_id,
+                project_id=self.project_id,
+                zone=self.zone,
+            )
+            self.log.info("The %s Instance already exists", self.resource_id)
+            ComputeInstanceDetailsLink.persist(
+                context=context,
+                task_instance=self,
+                location_id=self.zone,
+                resource_id=self.resource_id,
+                project_id=self.project_id or hook.project_id,
+            )
+            return Instance.to_dict(existing_instance)
```

Comment:
If this part does not throw NotFound but should be executed if get_instance passed, maybe it should go to try **else**?

Reply:
Done